src/main/java/org/apache/maven/plugins/assembly/mojos/AbstractAssemblyMojo.java

```diff
@@ -430,6 +430,15 @@ public abstract class AbstractAssemblyMojo extends AbstractMojo implements Assem
     @Parameter
     private String overrideGroupName;
 
+    /**
+     * Override of mask which is applied to permissions of files/directories before they are put into assembly.
+     * If {@code null} then the mask is not explicitly configured and remains implementation-specific.
+     * If invalid value is specified - like negative value - then behaviour is implementation specific, i.e. depends
+     * on underlying library which is used for building of assembly.
+     */
+    @Parameter
+    private Integer overrideUmask;
```

Comment:
Does this actually override a umask or does it set a umask? In the other cases above — GID, username — these are set on the files and may vary from one file to the next.  umask, however, is not a per-file property. It's applied to the permissions of the file (which is a per-file property)

Reply:
It overrides umask which otherwise is implementation specific (can be set by Plexus Archiver under some conditions or can be undefined). Refer to that [comment](https://github.com/apache/maven-assembly-plugin/pull/148#discussion_r1254863967) for details, please.

Reply:
Renamed `overrideUmask` into `umask` to avoid confusion.